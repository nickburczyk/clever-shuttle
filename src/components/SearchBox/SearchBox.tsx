import { useState } from "react"
import { useNavigate } from 'react-router-dom'
import styles from './searchBox.module.css'
import cn from 'classnames'

const { wrapper, input, searchBtn } = styles

const SearchBox = () => {
    const nav = useNavigate()
    const [query, setQuery] = useState('')

    const handleChange = (e: {target:HTMLInputElement}): void => {
        const { value } = e.target
        setQuery(value)
    }

    const handleSearch = () => {
        nav(`./profile/${query}`)
    }

    return (
        <div className={wrapper}>
            <input 
                className={input} 
                placeholder="Search by ID" 
                value={query} 
                onChange={e => handleChange(e)}
            />
            <button 
                className={cn('btn primary', searchBtn)}
                onClick={() => handleSearch()}
            >
                GO
            </button>
        </div>
    )
}

export default SearchBox