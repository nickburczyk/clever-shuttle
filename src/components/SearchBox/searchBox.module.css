.wrapper {
    margin-top: 1rem;
}

.input {
    padding: 1rem 1.5rem;
    border-radius: 4px 0 0 4px;
    border: 1px solid #d8dddb;
    border-right: none;
}

.searchBtn {
    border-radius: 0 4px 4px 0;
    border: 1px solid #00C763
}