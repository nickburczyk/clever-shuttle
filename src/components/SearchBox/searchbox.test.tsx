import React from "react";
import {create, act} from 'react-test-renderer'
import { Router } from 'react-router-dom'
import {createMemoryHistory} from 'history'

import SearchBox from "./SearchBox";

describe("Search Box", () => {
    const history = createMemoryHistory({ initialEntries: ['/'] });
    
    const tree = create(
        <Router navigator={history} location={history.location}>
            <SearchBox/>
        </Router>
    )

    it ("should render", () => {
        expect(tree.toJSON()).toMatchSnapshot()
    }) 
})