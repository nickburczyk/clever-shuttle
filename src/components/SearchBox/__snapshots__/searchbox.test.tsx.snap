// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Box should render 1`] = `
<div
  className="wrapper"
>
  <input
    className="input"
    onChange={[Function]}
    placeholder="Search by ID"
    value=""
  />
  <button
    className="btn primary searchBtn"
    onClick={[Function]}
  >
    GO
  </button>
</div>
`;
